tests/test_unit.py

```diff
@@ -551,28 +627,72 @@ def test_extract_header_from_markdown(self):
         # The header should be the same even with the license header.
         header_line_with_license_want = header_line_want
 
-        mdfile = open('tests/markdown_example_header.md', 'r')
+        mdfile_license = open('tests/markdown_example_header.md', 'r')
+        header_line_with_license_got = extract_header_from_markdown(mdfile_license)
+
+        self.assertEqual(header_line_with_license_got, header_line_with_license_want)
+        mdfile.close()
+
+
+    def test_extract_header_from_markdown_alternate_header(self):
+        # Check the header for an alternate header style.
+        header_line_want = "This is a simple alternate header"
+
+        mdfile = open('tests/markdown_example_alternate.md', 'r')
+        header_line_got = extract_header_from_markdown(mdfile)
+
+        self.assertEqual(header_line_got, header_line_want)
+        mdfile.close()
+
+        # The header should be the same even with the license header.
+        header_line_with_license_want = header_line_want
+
+        mdfile = open('tests/markdown_example_alternate_header.md', 'r')
         header_line_with_license_got = extract_header_from_markdown(mdfile)
 
         self.assertEqual(header_line_with_license_got, header_line_with_license_want)
         mdfile.close()
 
+        # Check the header for an alternate header style.
+        header_line_want = "This is a simple alternate header"
 
-    def test_extract_header_from_markdown_check_error(self):
-        # Check an exception is thrown for markdown files that's not well
-        # formatted.
-        mdfile = open('tests/markdown_example_bad.md', 'r')
-        with self.assertRaises(ValueError):
-            header_line = extract_header_from_markdown(mdfile)
+        mdfile = open('tests/markdown_example_alternate_less.md', 'r')
+        header_line_got = extract_header_from_markdown(mdfile)
 
+        self.assertEqual(header_line_got, header_line_want)
         mdfile.close()
 
-        mdfile = open('tests/markdown_example_noheader.md', 'r')
-        with self.assertRaises(ValueError):
-            header_line = extract_header_from_markdown(mdfile)
+
+    def test_extract_header_from_markdown_bad_headers(self):
+        # Check that the filename is used as header if no valid header is found.
+        header_line_want = "Markdown_example_bad_header"
+
+        mdfile = open('tests/markdown_example_bad_header.md', 'r')
+        header_line_got = extract_header_from_markdown(mdfile)
+
+        self.assertEqual(header_line_want, header_line_got)
 
         mdfile.close()
 
+        # Check that only h1 headers are parsed.
+        header_line_want = "Markdown_example_h2"
+
+        mdfile = open('tests/markdown_example_h2.md', 'r')
+        header_line_got = extract_header_from_markdown(mdfile)
+
+        self.assertEqual(header_line_want, header_line_got)
+
+        mdfile.close()
```

Comment:
This line can be removed if `mdfile = open('tests/markdown_example_h2.md', 'r')` is changed to a with statement. For example, `with open('tests/markdown_example_h2.md', 'r') as mdfile:`

Reply:
Thank you for the tip! Updating the files.